hazelcast/src/main/java/com/hazelcast/config/Config.java

```diff
@@ -173,10 +178,23 @@ public Config setClassLoader(ClassLoader classLoader) {
         return this;
     }
 
+    /**
+     * Returns the pattern matcher which is used to match item names to
+     * configuration objects.
+     *
+     * @return the pattern matcher
+     */
     public ConfigPatternMatcher getConfigPatternMatcher() {
         return configPatternMatcher;
     }
 
+    /**
+     * Sets the pattern matcher which is used to match item names to
+     * configuration objects.
+     *
```

Comment:
I would refrain from adding examples since the examples depend on the pattern matcher. I'll add the default value here.

Reply:
I don't even want to start on what should be in the next level of configs. In this PR we *only* want to fix the `Config` class.

Reply:
The next level of configs is doubly-hard to write since you need to go much deeper into the code to understand what each property is doing. And this isn't something we can do easily do.